auth/src/main/java/org/aerogear/auth/credentials/OIDCCredentials.java

```diff
@@ -32,6 +47,69 @@ public String getRefreshToken() {
+     * @return <code>true</code> if the token integrity is good.
+     */
+    public boolean verifyToken(final String jwtToken) {
+        String issuer = integrityCheckParameters.getIssuer();
+        String audience = integrityCheckParameters.getAudience();
+        String publicKey = integrityCheckParameters.getPublicKey();
+        return verifyToken(jwtToken, publicKey, issuer, audience);
+    }
+
+    /**
+     * A function to verify the authenticity of a JWT token against a public key, expected issuer and audience.
+     * @param jwtToken - The JWT Token to Verify.
+     * @param publicKey - The Public Key from Keycloak, without the Begin/End tags.
+     * @param issuer - The expected Issuer of the JWT
+     * @param audience - The expected Audience of the JWT
+     * @return <code>true</code> if the token integrity is good.
+     */
+    public boolean verifyToken(final String jwtToken, final String publicKey, final String issuer, final String audience) {
+        // add the Begin/End tags to the public key generated from Keycloak
+        String beginPublicKey = "-----BEGIN PUBLIC KEY-----";
+        String endPublicKey = "-----END PUBLIC KEY-----";
+        String constructedPublicKey = beginPublicKey + publicKey + endPublicKey;
+
+        // Convert the public key from a string to a Java security key
+        RsaKeyUtil utils = new RsaKeyUtil();
+        PublicKey jwtPublicKey = null;
+        try {
+            jwtPublicKey = utils.fromPemEncoded(constructedPublicKey);
+        } catch (JoseException e) {
+            e.printStackTrace();
+        } catch (InvalidKeySpecException e) {
+            e.printStackTrace();
+        }
+
+
+        // Validate and process the JWT.
+        JwtConsumer jwtConsumer = new JwtConsumerBuilder()
+                .setRequireExpirationTime() // require the JWT to have an expiration time
+                .setAllowedClockSkewInSeconds(30) // allow some leeway in validating time based claims to account for clock skew
+                .setRequireSubject() // require the subject claim
+                .setExpectedIssuer(issuer) // whom the JWT needs to have been issued by
+                .setExpectedAudience(audience) // to whom the JWT is intended for
+                .setVerificationKey(jwtPublicKey) // verify the signature with the public key
+                .setJwsAlgorithmConstraints( // only allow the expected signature algorithm(s) in the given context
+                        new AlgorithmConstraints(AlgorithmConstraints.ConstraintType.WHITELIST, // which is only RS256 here
+                                AlgorithmIdentifiers.RSA_USING_SHA256))
+                .build(); // create the JwtConsumer instance
+
+        try {
+            //  Validate the JWT and process it to the Claims
+            JwtClaims jwtClaims = jwtConsumer.processToClaims(jwtToken);
+            return true;
+        } catch (InvalidJwtException e) {
+            e.printStackTrace();
```

Comment:
If we can safely return `false` here, it would be better to use logger to log exceptions.